tests/github.test.ts

```diff
@@ -0,0 +1,64 @@
+import { mocked } from 'ts-jest/utils';
+
+import * as github from '@actions/github';
+import * as exec from '@actions/exec';
+jest.mock('@actions/exec');
+const mockedExec = mocked(exec, true);
+
+// Shallow clone original @actions/github context
+const originalContext = { ...github.context };
+const originalGhToken = process.env['GITHUB_TOKEN'];
+// let originalCwd = process.cwd();
+
+import * as common from '../src/common';
+jest.mock('../src/common');
+const mockedCommon = common as jest.Mocked<typeof common>;
+
+import fixtureGithubContext from './fixtures/github-context.json';
+import { Repo } from '../src/github';
+
+beforeEach(() => {
+  // Mock token env variable
+  process.env['GITHUB_TOKEN'] = 'gh-12abc';
+
+  // Mock the @actions/github context.
+  Object.defineProperty(github, 'context', {
+    value: fixtureGithubContext,
+  });
+
+  // process.chdir("./tests/fixtures");
+
+  mockedExec.exec.mockReset();
+});
+
+afterEach(() => {
+  // Restore original @actions/github context
+  Object.defineProperty(github, 'context', {
+    value: originalContext,
+  });
+  // Restore any original GITHUB_TOKEN env var
+  process.env['GITHUB_TOKEN'] = originalGhToken;
+
+  // Restore original CWD
+  // process.chdir(originalCwd);
+});
+
+test('getBasefile function', async () => {
+  // As we don't do any git operations in tests, we mock the resulting file
+  mockedCommon.fsExists.mockResolvedValue(true);
+
+  const repo = new Repo();
+  const previousFile = await repo.getBaseFile('openapi.yml');
+  const baseSha = fixtureGithubContext.payload.pull_request.base.sha;
+  const headSha = fixtureGithubContext.payload.pull_request.head.sha;
+  const baseBranch = '';
+
+  // Expect git executions
+  expect(mockedExec.exec.mock.calls).toEqual([
+    ['git', ['fetch', 'origin', baseSha, headSha]],
+    ['git', ['merge-base', baseSha, headSha], { listeners: expect.anything() }],
+    ['git', ['--work-tree', 'tmp/', 'restore', '-s', baseBranch, '.']],
+    ['git', ['restore', '-s', headSha, '.']],
+  ]);
+  expect(previousFile).toBe('tmp/openapi.yml');
```

Comment:
Why do we expect this?

Reply:
Because we expect the result of the `getBaseFile` to return the path of the base file

Reply:
Yes, I can see that, but I was wondering how this expectation was related to this change? I'm probably missing something, but I don't see a direct link here.

Reply:
It's not directly linked to the change. But as there were 0 tests on this function, it's normal to see expectations that are not related to the bugfix.

The expectations related to the bug fix are the `'.'` parameters givent to `git restore`